examples/loader-rust-to-json/src/index.md

```diff
@@ -0,0 +1,247 @@
+            *acc.entry(card.rank).or_insert(0) += 1;
+            acc
+        });
+        let suit_counts = self.0.iter().fold(HashMap::new(), |mut acc, card| {
+            *acc.entry(card.suit).or_insert(0) += 1;
+            acc
+        });
+        let is_flush = suit_counts.len() == 1;
+        let is_straight = if self.0.iter().any(|card| card.rank == 1) {
+            // Handle aces
+            let min_rank = self.0.iter().map(|card| card.rank).filter(|&rank| rank != 1).min().unwrap();
+            let max_rank = self.0.iter().map(|card| card.rank).filter(|&rank| rank != 1).max().unwrap();
+            (min_rank == 2 && max_rank == 5) || (min_rank == 10 && max_rank == 13)
+        } else {
+            let min_rank = self.0.iter().map(|card| card.rank).min().unwrap();
+            let max_rank = self.0.iter().map(|card| card.rank).max().unwrap();
+            (max_rank - min_rank) as usize == self.0.len() - 1
+        };
+
+        if is_flush && is_straight {
+            HandCategory::StraightFlush
+        } else if rank_counts.values().any(|&count| count == 4) {
+            HandCategory::FourOfAKind
+        } else if rank_counts.values().any(|&count| count == 3)
+            && rank_counts.values().any(|&count| count == 2)
+        {
+            HandCategory::FullHouse
+        } else if is_flush {
+            HandCategory::Flush
+        } else if is_straight {
+            HandCategory::Straight
+        } else if rank_counts.values().any(|&count| count == 3) {
+            HandCategory::ThreeOfAKind
+        } else if rank_counts.values().filter(|&&count| count == 2).count() == 2 {
+            HandCategory::TwoPair
+        } else if rank_counts.values().any(|&count| count == 2) {
+            HandCategory::OnePair
+        } else {
+            HandCategory::HighCard
+        }
+    }
+}
+```
+
+<div class="note">
+
+To run this data loader, you’ll need [rust-script](https://rust-script.org/) installed.
+
+</div>
+
+The above data loader lives in [`data/poker.json.rs`](data/poker.json.rs), so we can load the data as `data/poker.json` using `FileAttachment`.
+
+```js echo
+const hands = FileAttachment("data/poker.json").json();
+```
+
+```js echo
+const handsTs = FileAttachment("data/poker-ts.json").json();
+```
+
+<div style="max-width: 300px;">
+
+```js
+Inputs.table(hands.summary)
+```
+
+```js
+const extrapolatedTsTime = handsTs.meta.duration_ms * (hands.meta.count / handsTs.meta.count);
+```
+
+</div>
+
+Taking advantage of Rust’s performance and simple parallelism, the data loader above was able to summarize ${hands.meta.count / 1e6} million hands in ${hands.meta.duration_ms.toLocaleString("en-US")} ms. For comparison, an equivalent data loader written in TypeScript (which you can see in `data/poker-ts.json.ts`) would take ${Math.floor(extrapolatedTsTime / 1000).toLocaleString("en-US")} seconds to do the same simulation.
+
+We can make a quick chart of the poker hands with Plot
```

Comment:
Tiny nit, add a colon at the end of "We can make a quick chart of the poker hands with Plot"?

Reply:
```suggestion
We can make a quick chart of the poker hands with Plot:
```